### DISCUSSION TITLE
Question about keycloak integration in client basis

### DISCUSSION THREAD

**Discussion by Gianniskourentzis** [stellio-hub/stellio-context-broker#1128] [Q&A] [answered]
Hello folks,
I am the stellio context broker and also easyglobalmarket/keycloak docker images, providing by fiware for my project. I think that i have set it up correctly, and i can now login in Keycloak UI. 
By following, Fiware's tutorial ([keycloak integration](https://stellio.readthedocs.io/en/latest/admin/keycloak_integration.html)), I realise that keycloak secures stellio API on cilent basis (by providing client secret and id). 
Is there any way, to secure stellio API on user basis? By this i mean, is there any way to use, user's username and password to have an access token and access API? 
If not, is there any way to implement this? 
Thanks

I tried the whole proccess from tutorial but users credentials not do anything!

**Comment by bobeal** [comment, marked as answer]
Hi @Gianniskourentzis,

AFAIK, it is not something that is possible with an OpenID Connect / OAuth2 protected API (at least not without implying a client application).

You will find here the different grant types that are supported by Keycloak (I guess it supports all known ones): https://www.keycloak.org/docs/latest/securing_apps/#supported-grant-types. You can also find some more info here: https://www.oauth.com/oauth2-servers/access-tokens/
